### PR TITLE
Fix CI out of memory & add PyTorch1.9 Python3.9 unit tests

### PR DESCRIPTION
**Pull request by RangiLyu**
# Motivation
Unit tests are always killed because out of memory. This PR reduces memory footprint and skips some heavy tests. 
Also support PyTorch1.9 and Python3.9 in this PR.

# Modification
- Skip tests of building all configs.
- Reduce base_channels of resnet in most of the tests.
- Reduce the input size.
- Delete some redundant tests.

### CODE REVIEW

File: tests/test_models/test_backbones/test_csp_darknet.py [hunk 1]
```diff
@@ -35,90 +34,84 @@ def test_csp_darknet_backbone():
...
-    assert feat[0].shape == torch.Size((1, 64, 112, 112))
-    assert feat[1].shape == torch.Size((1, 128, 56, 56))
-    assert feat[2].shape == torch.Size((1, 256, 28, 28))
-    assert feat[3].shape == torch.Size((1, 512, 14, 14))
-    assert feat[4].shape == torch.Size((1, 1024, 7, 7))
+    assert feat[0].shape == torch.Size((1, 64, 32, 32))
+    assert feat[1].shape == torch.Size((1, 128, 16, 16))
+    assert feat[2].shape == torch.Size((1, 256, 8, 8))
+    assert feat[3].shape == torch.Size((1, 512, 4, 4))
+    assert feat[4].shape == torch.Size((1, 1024, 2, 2))
```
Comment (ZwwWayne) [inline tests/test_models/test_backbones/test_csp_darknet.py:54]: We can try to further reduce the number of channels

File: tests/test_models/test_backbones/test_hourglass.py [hunk 2]
```diff
@@ -24,22 +24,26 @@ def test_hourglass_backbone():
...
     # Test HourglassNet-52
-    model = HourglassNet(num_stacks=1)
-    model.init_weights()
+    model = HourglassNet(
+        num_stacks=1,
+        stage_channels=(64, 64, 96, 96, 96, 128),
+        feat_channel=64)
     model.train()
 
     imgs = torch.randn(1, 3, 256, 256)
```
Comment (ZwwWayne) [inline tests/test_models/test_backbones/test_hourglass.py:33]: can we make it as 128?

File: tests/test_models/test_backbones/test_resnet.py [hunk 3]
```diff
@@ -464,61 +456,57 @@ def test_resnet_backbone():
...
     feat = model(imgs)
     assert len(feat) == 4
-    assert feat[0].shape == torch.Size([1, 256, 56, 56])
-    assert feat[1].shape == torch.Size([1, 512, 28, 28])
-    assert feat[2].shape == torch.Size([1, 1024, 14, 14])
-    assert feat[3].shape == torch.Size([1, 2048, 7, 7])
+    assert feat[0].shape == torch.Size([1, 256, 8, 8])
+    assert feat[1].shape == torch.Size([1, 512, 4, 4])
+    assert feat[2].shape == torch.Size([1, 1024, 2, 2])
+    assert feat[3].shape == torch.Size([1, 2048, 1, 1])
```
Comment (ZwwWayne) [inline tests/test_models/test_backbones/test_resnet.py:467]: May also try to reduce the base channels

**Comment by ZwwWayne** [comment]
We can also consider reducing the size of the feature map in test_dense_heads.

File: tests/test_models/test_backbones/test_hourglass.py (hunk 2 above)
Comment (RangiLyu) [reply tests/test_models/test_backbones/test_hourglass.py:33]: The inner block will has stride 256

**Comment by RangiLyu** [comment]
All checks passed. Please take a look. @ZwwWayne

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -185,3 +159,68 @@ jobs:
...
+        if: ${{matrix.torchvision < 0.5}}
+      - name: Install PyTorch
+        run: python -m pip install torch==${{matrix.torch}} torchvision==${{matrix.torchvision}} -f https://download.pytorch.org/whl/torch_stable.html
+      - name: Install dependencies for compiling onnx when python=3.9
+        run: python -m pip install protobuf && apt-get update && apt-get -y install libprotobuf-dev protobuf-compiler cmake
+        if: ${{matrix.python-version == '3.9-dev'}}
+      - name: Install mmdet dependencies
+        run: |
+          python -V
+          python -m pip install mmcv-full -f https://download.openmmlab.com/mmcv/dist/cu101/${{matrix.mmcv_link}}/index.html
```
Comment (ZwwWayne) [inline .github/workflows/build.yml:203]: should we use cu102 here?

**Review by ZwwWayne — APPROVED** [review]

**Pull request merged**